Activities/Com.weboniselab.hemant.android.activities/app/src/main/java/com/example/webonise/comweboniselabhemantandroidactivities/MainActivity.java

```diff
@@ -0,0 +1,118 @@
+package com.example.webonise.comweboniselabhemantandroidactivities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.PersistableBundle;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+
+public class MainActivity extends Activity implements View.OnClickListener {
+    EditText edUsername;
+    EditText edPassword;
+    Button btnValidate;
+
+    protected void initialize() {
+        edUsername = (EditText) findViewById(R.id.et_username);
+        edPassword = (EditText) findViewById(R.id.et_password);
+        btnValidate = (Button) findViewById(R.id.btn_Validation);
+        check();
+//      Very Important We have to setOnclickListner on the btn.
+        btnValidate.setOnClickListener(this);
+
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Log.w("1","onCreate");
+        setContentView(R.layout.activity_main);
+        initialize();
+
+    }
+
+
+    private void check() {
+        String uname = edUsername.getText().toString();
+        String pass = edPassword.getText().toString();
+
+        if (TextUtils.isEmpty(uname)) {
+            edUsername.setError("Username-empty");
```

Comment:
use of setError is good. But why you did not use string.xml for this. And Error message should be little meaningful